json-schema-core/src/main/java/io/ballerina/jsonschema/core/Generator.java

```diff
@@ -76,6 +84,12 @@ public class Generator {
     private final ArrayList<String> imports = new ArrayList<>();
     private final List<JsonSchemaDiagnostic> diagnostics = new ArrayList<>();
 
+    private int constCounter = 1;
+
+    public void setConstCounter(int newCount) {
```

Comment:
What we discussed was making this something like getNextIndex. This method will then always increment and give the next index.